TrackerGeom/inc/Tracker.hh

```diff
@@ -15,70 +16,133 @@
 #include <array>
 #include <limits>
 #include <string>
+#include <memory>
 
 #include "cetlib_except/exception.h"
 
-
 #include "Mu2eInterfaces/inc/Detector.hh"
 #include "Mu2eInterfaces/inc/ProditionsEntity.hh"
 #include "TrackerGeom/inc/Manifold.hh"
 #include "TrackerGeom/inc/Support.hh"
 #include "TrackerGeom/inc/SupportModel.hh"
 #include "TrackerGeom/inc/SupportStructure.hh"
+#include "GeneralUtilities/inc/HepTransform.hh"
 
 #include "TrackerGeom/inc/Plane.hh"
-#include "DataProducts/inc/PanelId.hh"
+#include "TrackerGeom/inc/Panel.hh"
 #include "GeomPrimitives/inc/TubsParams.hh"
 #include "GeomPrimitives/inc/PlacedTubs.hh"
+#include "TrackerGeom/inc/PanelEB.hh"
+#include "TrackerGeom/inc/StrawProperties.hh"
 
 namespace mu2e {
-
   class Tracker : public Detector, public ProditionsEntity {
+    friend class TrackerMaker; // remove after factorizing out the G4 stuff FIXME!
+    using xyzVec = CLHEP::Hep3Vector; // switch to XYZVec TODO
 
-        friend class TrackerMaker;
-        friend class AlignedTrackerMaker;
-
-  public:
-
+    public:
     typedef std::shared_ptr<Tracker> ptr_t;
     typedef std::shared_ptr<const Tracker> cptr_t;
+    using PlaneCollection = std::array<Plane,StrawId::_nplanes>;
+    using PanelCollection = std::array<Panel,StrawId::_nupanels>;
+    using StrawCollection = std::array<Straw,StrawId::_nustraws>;
+    using StrawIndexMap = std::array<uint16_t,StrawId::_maxval>; 
+    // default constructor results in non-functional object, but is required by proditions service
```

Comment:
I don't recall why Proditions requires a default constructor, can you point it out?

Reply:
see above.